smac/intensifier/abstract_intensifier.py

```diff
@@ -571,8 +571,12 @@ def update_incumbents(self, config: Configuration) -> None:
 
         if len(previous_incumbents) == len(new_incumbents):
             if previous_incumbents == new_incumbents:
-                # No changes in the incumbents
-                self._remove_rejected_config(config_id)
+                # No changes in the incumbents, we need this clause because we can't use set difference then
```

Comment:
However, since all the configs are unique and disordered in `_rejected_config_ids`,  I cannot find a reason not to store `config_ids` as Set()